src/Controls/src/Build.Tasks/CompiledConverters/BrushTypeConverter.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Maui.Controls.Build.Tasks;
+using Microsoft.Maui.Controls.Xaml;
+using Microsoft.Maui.Graphics;
+using Mono.Cecil.Cil;
+
+namespace Microsoft.Maui.Controls.XamlC;
+
+class BrushTypeConverter : ICompiledTypeConverter
+{
+	public IEnumerable<Instruction> ConvertFromString(string value, ILContext context, BaseNode node)
+	{
+		var module = context.Body.Method.Module;
+
+		do
+		{
+			if (string.IsNullOrEmpty(value))
+				break;
+
+			value = value.Trim();
+
+			if (value.StartsWith("#", StringComparison.Ordinal))
+			{
+				var color = Color.FromArgb(value);
+
+				yield return Instruction.Create(OpCodes.Ldc_R4, color.Red);
+				yield return Instruction.Create(OpCodes.Ldc_R4, color.Green);
+				yield return Instruction.Create(OpCodes.Ldc_R4, color.Blue);
+				yield return Instruction.Create(OpCodes.Ldc_R4, color.Alpha);
+
+				yield return Instruction.Create(OpCodes.Newobj, module.ImportCtorReference(("Microsoft.Maui.Graphics", "Microsoft.Maui.Graphics", "Color"), parameterTypes: new[] {
+						("mscorlib", "System", "Single"),
+						("mscorlib", "System", "Single"),
+						("mscorlib", "System", "Single"),
+						("mscorlib", "System", "Single")}));
+
+				yield return Instruction.Create(OpCodes.Newobj, module.ImportCtorReference(("Microsoft.Maui.Controls", "Microsoft.Maui.Controls", "SolidColorBrush"), parameterTypes: new[] {
+						("Microsoft.Maui.Graphics", "Microsoft.Maui.Graphics", "Color")}));
+
+				yield break;
+			}
+
+			var parts = value.Split('.');
+			if (parts.Length == 1 || (parts.Length == 2 && parts[0] == "Brush"))
```

Comment:
I don't think the non-compiled converter supports `Brush.Red` syntax, so no need to add it here

Reply:
Do we need `Color.Red`, here: 

https://github.com/dotnet/maui/blob/e38582b53564b85398588e9ecaba43e325fa04fa/src/Controls/src/Build.Tasks/CompiledConverters/ColorTypeConverter.cs#L39-L40

Reply:
I'd gladly get rid of `Color.` prefix, but that depends on how much people we break. not adding `Brush.` won't break anyone.

What we can do at first, is send a warning when using `Color.Red` saying 'this can be simplified. There's a now a precedent, and since a few weeks the Xaml compiler can send warning in addition of exceptions